src/zarr/codecs/pipeline/batched.py

```diff
@@ -0,0 +1,281 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, TypeVar, Iterable
+import numpy as np
+from dataclasses import dataclass
+
+from zarr.abc.codec import (
+    ByteGetter,
+    ByteSetter,
+    Codec,
+    ArrayArrayCodec,
+    ArrayBytesCodec,
+    ArrayBytesCodecPartialDecodeMixin,
+    ArrayBytesCodecPartialEncodeMixin,
+    BytesBytesCodec,
+)
+from zarr.codecs.pipeline.core import CodecPipeline
+from zarr.common import concurrent_map
+from zarr.indexing import is_total_slice
+
+if TYPE_CHECKING:
+    from typing import List, Optional, Tuple
+    from zarr.metadata import RuntimeConfiguration
+    from zarr.common import ArraySpec, BytesLike, SliceSelection
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+def _unzip2(iterable: Iterable[Tuple[T, U]]) -> Tuple[List[T], List[U]]:
```

Comment:
what's the difference between this function and zip with tuple unpacking (the last line in my example)?

```python
>>> a = ('a','b','c')
>>> b = (1, 2, 3)
>>> tuple(zip(a,b))
(('a', 1), ('b', 2), ('c', 3))
>>> z = tuple(zip(a,b))
>>> tuple(zip(*z))
(('a', 'b', 'c'), (1, 2, 3))
```

Reply:
Functionally equivalent, but mypy didn't understand the types. I got `Tuple[List[T | U], ...]` as return type.